### ISSUE TITLE
invalid peer dependency warnings

### ISSUE THREAD

**Issue by dadamssg** [yarnpkg/yarn#4531]
Yarn: 1.0.2
Node: 8.05
Mac

Starting with a brand new project, I installed `standard`. This cause several peer dependency warnings:

```
warning "eslint-plugin-react@6.10.3" has incorrect peer dependency "eslint@^2.0.0 || ^3.0.0".
warning "eslint-plugin-import@2.2.0" has incorrect peer dependency "eslint@2.x - 3.x".
warning "eslint-config-standard-jsx@4.0.2" has incorrect peer dependency "eslint@>=3.19.0".
warning "eslint-config-standard-jsx@4.0.2" has incorrect peer dependency "eslint-plugin-react@>=6.10.3".
warning "eslint-plugin-node@4.2.3" has incorrect peer dependency "eslint@>=3.1.0".
warning "eslint-config-standard@10.2.1" has incorrect peer dependency "eslint@>=3.19.0".
warning "eslint-config-standard@10.2.1" has incorrect peer dependency "eslint-plugin-import@>=2.2.0".
warning "eslint-config-standard@10.2.1" has incorrect peer dependency "eslint-plugin-node@>=4.2.2".
warning "eslint-config-standard@10.2.1" has incorrect peer dependency "eslint-plugin-promise@>=3.5.0".
warning "eslint-config-standard@10.2.1" has incorrect peer dependency "eslint-plugin-standard@>=3.0.0".
warning "eslint-plugin-standard@3.0.1" has incorrect peer dependency "eslint@>=3.19.0".
warning "ajv-keywords@1.5.1" has incorrect peer dependency "ajv@>=4.10.0".
```

Doing this should not cause any peer dependency warnings. For example, eslint `3.19.0` gets installed which satisfies the eslint warnings above. 

After looking at the warnings and seeing standard's package.json's file depending on `eslint@~3.19.0`, I thought I could get rid of the eslint warnings by `yarn add eslint@~3.19.0 --dev`. I thought maybe yarn was inspecting my project's package.json file for eslint. This did not suppress the warnings. Doing `yarn add eslint@^3.19.0 --dev` did resolve the warnings. This doesn't make sense. Having `^3.19.0` in my package.json file would allow `3.20.0` to be installed(if it existed) but that would be out of standard's `~3.19.0` range.

**Comment by bartdominiak** [comment]
It's an issue in yarn@1.0.2. with peerDependency
Use [nightly](https://yarnpkg.com/en/docs/nightly) to update your yarn to newest version.

**Comment by BYK** [comment]
Closing since this is fixed on master and with 1.1.0
